### ISSUE TITLE
Determine candidate alleles

### ISSUE THREAD

**Issue by cyycyj** [sc-zhang/AlleleFinder#7]
Dear Shengcheng,

First of all, I would like to express my appreciation for these epic and stable tools – they run smoothly and quickly.

On my occasion, my genome is diploid, and I have tried using a haplotype as the reference gene. Here is my AlleleFinder script:

```
#!/bin/bash
#SBATCH --job-name=01.AlleleFinder
#SBATCH --partition tPartition
#SBATCH --cpus-per-task=30
#SBATCH --ntasks-per-node=1
#SBATCH -o 01.AlleleFinder.%j.out
#SBATCH -e 01.AlleleFinder.%j.err
#SBATCH --nodes=1

source ~/.bashrc

source activate allelefinder

genome1=Lind.hap1
genome2=Lind.hap2

cat $genome1.modified.cds.fa $genome2.modified.cds.fa > merge.cds.fa &&
cat $genome1.modified.gff3 $genome2.modified.gff3 > merge.gff3 &&

AlleleFinder -r $genome1.modified.fa \
-d $genome1.modified.cds.fa \
-f $genome1.modified.gff3 \
-c merge.cds.fa \
-g merge.gff3 \
-n 2 \
-w workdir \
-m \
-t 30
```

I have encountered the following situation:

```
#CHR    POS     ref gene        Allele A        Allele B
...
Chr01A  602834  LindH1C01G000078     LindH1C01G000076,LindH1C01G000078-P       LindH2C01G000077,LindH2C01G000079-P
Chr01A  612880  LindH1C01G000079     LindH1C01G000075,LindH1C01G000077-P,LindH1C01G000079-P LindH2C01G000076,LindH2C01G000078-P
...
Chr01A  5271055 LindH1C01G000731     LindH1C01G000616,LindH1C01G000731-P       LindH2C04G001875
```

As you can see, the ref gene LindH1C01G000731 has been assigned to LindH1C01G000616 in Allele A, while LindH1C01G000731 itself has been regarded as -P, which means paralog. I wonder which one I should choose as the alleles?

Looking forward to hearing from you.

**Comment by sc-zhang** [comment]
@cyycyj the main strategy we construct allele table is based on gene similarity, and the reference genome and annotation is used for anchoring genes as alleles.
In the allele table, the first three columns are worked as coordinate, and genes in same row means they are allelic genes. Genes which are in the same row and column means they are identified as allelic genes and located on the same hap,  then we will choose the first one (ordered by the lexicographic) as allele gene (without -P/-T), and the others as paralog (-P) or tandem (-T, identified by MCScanX). That was why the gene itsself may be regarded as paralog when use one hap as reference.
In conclusion, we do not furthur differentiate gene and its paralog and/or tandem on this occassion.

**Comment by cyycyj** [comment]
Dear Shengcheng,

Thank you for your quick response. For my situation, should I choose the allele gene based on the reference gene column (opting for LindH1C01G000731 over LindH1C01G000616, as in my previous example)? Also, I'd like to know how to select the right alleles in the following example, especially when we face a multiple versus multiple candidates or a multiple versus one condition scenario?

```
#CHR    POS     ref gene        Allele A        Allele B

#multiple vs multiple
Chr03A  17215966        LindH1C03G000950     LindH1C02G000957,LindH1C03G000950-P,LindH1C01G003125-P LindH2C12G000466,LindH2C15G
000994-P,LindH2C02G000938-P,LindH2C10G001014-P,LindH2C10G001022-P,LindH2C04G001674-P,LindH2C04G001679-P

#mutiple vs one
Chr01A  17492138        LindH1C01G002267     LindH1C01G002266,LindH1C01G002267-T       LindH2C01G002248
```

By the way, considering the output, maybe we can use this tool to identify gene clusters.

Cheers

**Comment by sc-zhang** [comment]
For the first question, you can do that, and for the second one, in my opinion, if you care about it, you can blast genes between allele A and B, then chose the most similar two genes as allelic genes, and the others as paralog or tandem.
